proposals/testing/NNNN-polling-confirmations.md

```diff
@@ -0,0 +1,512 @@
+/// stop condition.
+///
+/// To add this trait to a test, use the ``Trait/pollingUntilStopsPassingDefaults``
+/// function.
+@available(_clockAPI, *)
+public struct PollingUntilStopsPassingConfigurationTrait: TestTrait, SuiteTrait {
+  /// How long to continue polling for
+  public var duration: Duration?
+  /// The minimum amount of time to wait between polling attempts
+  public var interval: Duration?
+
+  public var isRecursive: Bool { true }
+}
+
+@available(_clockAPI, *)
+extension Trait where Self == PollingUntilFirstPassConfigurationTrait {
+  /// Specifies defaults for ``confirmPassesEventually`` in the test or suite.
+  ///
+  /// - Parameters:
+  ///   - duration: The expected length of time to continue polling for.
+  ///     This value may not correspond to the wall-clock time that polling
+  ///     lasts for, especially on highly-loaded systems with a lot of tests
+  ///     running.
+  ///     if nil, polling will be attempted for approximately 1 second.
+  ///     `duration` must be greater than 0.
+  ///   - interval: The minimum amount of time to wait between polling
+  ///     attempts.
+  ///     If nil, polling will wait at least 1 millisecond between polling
+  ///     attempts.
+  ///     `interval` must be greater than 0.
+  public static func pollingUntilFirstPassDefaults(
+    until duration: Duration? = nil,
+    pollingEvery interval: Duration? = nil
+  ) -> Self
+}
+
+@available(_clockAPI, *)
+extension Trait where Self == PollingUntilStopsPassingConfigurationTrait {
+  /// Specifies defaults for ``confirmPassesAlways`` in the test or suite.
+  ///
+  /// - Parameters:
+  ///   - duration: The expected length of time to continue polling for.
+  ///     This value may not correspond to the wall-clock time that polling
+  ///     lasts for, especially on highly-loaded systems with a lot of tests
+  ///     running.
+  ///     if nil, polling will be attempted for approximately 1 second.
+  ///     `duration` must be greater than 0.
+  ///   - interval: The minimum amount of time to wait between polling
+  ///     attempts.
+  ///     If nil, polling will wait at least 1 millisecond between polling
+  ///     attempts.
+  ///     `interval` must be greater than 0.
+  public static func pollingUntilStopsPassingDefaults(
+    until duration: Duration? = nil,
+    pollingEvery interval: Duration? = nil
+  ) -> Self
+}
+```
+
+Specifying `duration` or `interval` directly on either new `confirmation`
+function will override any value provided by the relevant trait. Additionally,
+when multiple of these configuration traits are specified, the innermost or
+last trait will be applied.
+
+### Default Polling Configuration
+
+For all polling confirmations, the Testing library will default `duration` to
+1 second, and `interval` to 1 millisecond.
+
+### Platform Availability
+
+Polling confirmations will not be available on platforms that do not support
+Swift Concurrency.
```

Comment:
The availability will also be constrained to OS versions which have the `Clock`, `Duration`, and related types. You may want to mention the specific Apple OS versions in particular.